src/s3/commands/uploadFile.ts

```diff
@@ -104,104 +139,245 @@ export async function uploadFileCommand(
                 telemetry.recordS3UploadObject({ result: 'Cancelled' })
                 return
             }
         }
     }
-    const fileName = path.basename(file.fsPath)
-    const destinationPath = readablePath({ bucket: { name: bucket.Name! }, path: key })
 
-    try {
-        showOutputMessage(localize(
-            'AWS.s3.uploadFile.startUpload', 
-            'Uploading file {0} to {1}', 
-            fileName, 
-            destinationPath), 
-        outputChannel)
-
-        const request = {
-            bucketName: bucket.Name!,
-            key: key,
-            fileLocation: file,
-            fileSizeBytes: fileSizeBytes(file),
-            s3Client,
-            window: window,
-        }
+    let failedRequests = await uploadBatchOfFiles(uploadRequests, window, outputChannel)
 
-        await uploadWithProgress(request)
+    const completedRequests = uploadRequests.length - failedRequests.length
+    showOutputMessage(
+        localize(
+            'AWS.s3.uploadFile.success',
+            'Successfully uploaded {0}/{1} files',
+            completedRequests,
+            uploadRequests.length
+        ),
+        outputChannel
+    )
 
-        showOutputMessage(localize(
-            'AWS.s3.uploadFile.success', 
-            'Successfully uploaded file {0} to {1}', 
-            fileName, 
-            bucket.Name), 
-        outputChannel)
-        telemetry.recordS3UploadObject({ result: 'Succeeded' })
-        recordAwsRefreshExplorer()
-        commands.execute('aws.refreshAwsExplorer')
-        return
-    } catch (e) {
-        getLogger().error(`Failed to upload file from ${file} to ${destinationPath}: %O`, e)
-        showErrorWithLogs(localize('AWS.s3.uploadFile.error.general', 'Failed to upload file {0}', fileName), window)
-        telemetry.recordS3UploadObject({ result: 'Failed' })
-        return
+    while (failedRequests.length > 0) {
+        const failedKeys = failedRequests.map(request => request.key)
+        getLogger().error(`List of requests failed to upload:\n${failedRequests.toString().split(',').join('\n')}`)
+
+        if (failedRequests.length > 5) {
+            showOutputMessage(
+                localize(
+                    'AWS.s3.uploadFile.failedMany',
+                    'Failed uploads:\n{0}\nSee logs for full list of failed items',
+                    failedKeys.toString().split(',').slice(0, 5).join('\n')
+                ),
+                outputChannel
+            )
+        } else {
+            showOutputMessage(
+                localize(
+                    'AWS.s3.uploadFile.failed',
+                    'Failed uploads:\n{0}',
+                    failedKeys.toString().split(',').join('\n')
+                ),
+                outputChannel
+            )
+        }
+        const tryAgain = localize('AWS.generic.response.tryAgain', 'Try again')
+        const cont = localize('AWS.generic.response.continue', 'Continue')
```

Comment:
I think "Skip" is less confusing than "Continue" in this context. Also let's rename the other button to "Retry".